src/components/Calendar.tsx

```diff
@@ -69,10 +73,36 @@ const getMarkedDates = (data?: any[]) => {
 
 export const Calendar = ({ query, queryVariables, calendarData, isLoading, navigation }: Props) => {
   const contentContainerId = queryVariables?.contentContainerId;
+  const today = moment().format('YYYY-MM-DD');
+  const { globalSettings } = useContext(SettingsContext);
+  const { settings = {} } = globalSettings;
+  const { eventCalendar = {} } = settings;
+  const { maxDotPerDay: MAX_DOTS_PER_DAY = 5, subList = false } = eventCalendar;
+
+  const [queryVariableWithDateRange, setQueryVariableWithDateRange] = useState<any>({
+    queryVariables,
+    dateRange: [today, today]
+  });
+  const [markedDates, setMarkedDates] = useState<CalendarProps['markedDates']>({});
+
+  const { data, loading } = useQuery(getQuery(QUERY_TYPES.EVENT_RECORDS), {
+    variables: queryVariableWithDateRange
+  });
+
+  useEffect(() => {
+    setMarkedDates(getMarkedDates(calendarData, MAX_DOTS_PER_DAY, today));
+  }, []);
```

Comment:
couldn't this be the default for `const [markedDates, setMarkedDates] = useState<CalendarProps['markedDates']>({});` directly instead of having it in this effect here?

Reply:
Yes, you are right, it can be added by default. I am updating it.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/34045f488b0635c5c06dffbb0f7dc8b139a98275